test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -281,13 +281,18 @@ spec =
             Meta.get_annotation value "Value" "bar" . should_equal Nothing
             Meta.get_annotation value "Value" "baz" . should_equal (My_Type.Value 1 2 3)
 
-    Test.group "Check Nothing" <|
+    Test.group "Check Nothing and NaN" <|
         Test.specify "Nothing.is_a Nothing" <|
             Nothing.is_a Nothing . should_be_true
+            Meta.is_same_object Nothing Nothing . should_be_true
 
         Test.specify "type_of Nothing is Nothing" <|
             Meta.type_of Nothing . should_equal Nothing
 
+        Test.specify "NaN and NaN should be the same object" <|
```

Comment:
This is exactly how JS works, according to the spec in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#same-value-zero_equality